### ISSUE TITLE
Feat: Comment feature on posts

### ISSUE THREAD

**Issue by Sanchitbajaj02** [Sanchitbajaj02/palettegram#99]
### What feature?

Currently there is no feature to comment on the posts. Comment is way to express your ideas or criticisim towards the post so it is one of the important feature.
To create this, dev has to consider making an API. Appwrite backend is configured to accept comments in an array data structure but further discussion is needed.

### Record

- [X] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue

**Comment by sumanjeet0012** [comment]
I want to work on this issue.

I have previously worked on some Appwrite projects, however I will still require some help when needed.

**Comment by Sanchitbajaj02** [comment]
Sure @sumanjeet0012, me as well many mentors are here to help you on discord. Assigned this to you 😊

**Comment by sumanjeet0012** [comment]
I have worked on the comment feature on posts ,
The comments are getting stored on backend at appwrite.
but the `singlePost` ( in singlePost.tsx) is not updating by itself when a post is added on appwrite, it is updating after page reload.
any suggestions for this problem.

I have added the current progress video.

https://github.com/Sanchitbajaj02/palettegram/assets/78401409/cea7e5e5-1e66-45ee-970c-32f1605ce8af
